Draw toolbar UI [#175405075]

kswenson opened the pull request:
[[#175405075]](https://www.pivotaltracker.com/story/show/175405075)

Phase I
- Top-level buttons are converted to spec
- Stroke/fill color buttons bring up old settings dialog (for now)
- Sub-toolbars will be in the next phase
- Stamps button hasn't changed (yet)

<img width="335" alt="DrawToolbarPhaseI" src="https://user-images.githubusercontent.com/235234/98307310-57890200-1f7a-11eb-9782-0b9002fc938b.png">

Phase II
- Stroke and fill color buttons use subpalettes that appear below the toolbar (or above if they don’t fit below).
- Settings popup has been removed (no way to set stroke width or pattern).
- Stamp button/popup still behave as before (that’s phase III).

<img width="418" alt="DrawToolbarPhaseII" src="https://user-images.githubusercontent.com/235234/98616173-17df5480-22b1-11eb-8d14-1d68c07cbaae.png">

Phase III
- Stamps button follows updated UI spec
- Clicking the stamps button selects the stamp tool
- Tap-holding the stamps button shows/hides the stamps subpalette
- Clicking the small disclosure triangle in corner of button shows/hides the stamps subpalette without a hold delay

<img width="378" alt="DrawToolbarPhaseIII" src="https://user-images.githubusercontent.com/235234/98616225-35acb980-22b1-11eb-8454-1455ea1470e4.png">

mklewandowski commented on src/components/tools/drawing-tool/drawing-toolbar-buttons.tsx:
```diff
@@ -104,11 +105,12 @@ export const StampModeButton: React.FC<IStampModeButtonProps> = ({
+          <div className="expand-collapse" onClick={handleExpandCollapseClick}>
+            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 7 7" width="7" height="7">
+              <polygon points="0,7 7,7 7,0"/>
+            </svg>
```
Any reason not too store this in an SVG file that gets imported in case it's needed elsewhere?  Granted, it's so darn simple that that approach might be overkill.

mklewandowski replied:
I see there are a few of these cases where small SVGs are defined inline.  This seems like the one that might be reusable.

mklewandowski commented on src/components/tools/drawing-tool/color-swatch.tsx:
```diff
@@ -0,0 +1,46 @@
+  gray = "#bfbfbf",
+  orange = "#ff8415",
+  yellow = "#ffff00",
+  purple = "#d100d1"
```
I think most of these are defined in `vars.scss`.  Not sure it's worth the effort to import them from there.

kswenson replied:
I thought about it but never quite got up the motivation to do it. It probably would make sense, however, both for the colors and for some of the sizing constants.

mklewandowski commented on src/components/tools/drawing-tool/drawing-toolbar.tsx:
```diff
@@ -90,71 +106,51 @@ export const ToolbarView: React.FC<IProps> = (
-                onStrokeWidthChange={handleStrokeWidthChange} />
+          {paletteState.showStroke
+            ? <StrokeColorPalette selectedColor={drawingContent.stroke} onSelectColor={handleStrokeColorChange} />
+            : null}
```
A few of these could elimate the `null` and be condensed to just:
`{isFoo && <ReactComponent />}`

mklewandowski commented on src/utilities/color-utils.ts:
```diff
@@ -0,0 +1,17 @@
+  return (r * 0.2126 + g * 0.7152 + b * 0.0722) / scale;
+};
+
+export const luminanceColorString= (color: string) => {
```
missing space: `export const luminanceColorString = (color: string) => {`

mklewandowski commented on src/utilities/color-utils.ts:
```diff
@@ -0,0 +1,17 @@
+  const kLightLuminanceThreshold = 0.85;
+  const luminance = color && luminanceColorString(color);
+  return (luminance != null) && (luminance >= kLightLuminanceThreshold);
+};
```
Somewhat related (and part of a different discussion), but I had to build some similar color util functions in the Activity Player.  Maybe in the future these sorts of utils could be unified?  Just thinking out loud.
https://github.com/concord-consortium/activity-player/blob/master/src/utilities/theme-utils.ts

kswenson replied:
I think I looked at those but felt that they didn't seem quite on point for my use case. You might consider using the `color-string` library for AP so you don't have to do the parsing yourself.

mklewandowski replied:
Agree.  Just wondering if at some point we have a bunch of projects that have util functions like this and if they belong in some general purpose location (probably the translation work has me thinking in this direction).  Anyway, nothing here in CLUE needs to change.

mklewandowski reviewed (approved):
👍 LGTM!  I added a few comments, but the main bugs that I was seeing all appear to be resolved.

kswenson commented:
I added a commit which addressed some of your comments, including moving the `SmallCornerTriangle` SVG to its own file for potential reuse and getting rid if the extraneous ternaries.
